src/site/content/en/blog/pwa-url-handler/index.md

```diff
@@ -0,0 +1,291 @@
+associated URLs on Android at the operating system level using intent filters. {% endAside %}
+
+### The `"url_handlers"` Web App Manifest member
+
+To associate an installed PWA with URL patterns, these patterns need to be specified in the Web App
+Manifest. This happens through the `"url_handlers"` member. It accepts an array of objects with the
+following property:
+
+- `origin`: A required `string`, which is a pattern for matching origins. These patterns are allowed
+  to have a wildcard (`*`) prefix in order to include multiple sub-domains (like `*.example.com`).
+  URLs that match these origins could be handled by this web app. The scheme is always assumed to be
+  `https://`.
+
+The excerpt of a Web App Manifest below shows how the music PWA example from the introductory
+paragraph could set this up. The second entry with the wildcard (`"*.music.example.com"`) makes sure
+that the app also gets activated for `www.music.example.com` or potential other examples like
+`marketing-activity.music.example.com`.
+
+```json
+{
+  "url_handlers": [
+    {
+      "origin": "music.example.com"
+    },
+    {
+      "origin": "*.music.example.com"
+    },
+    {
+      "origin": "🎵.example.com"
+    }
+  ]
+}
+```
+
+{%Aside %} While in an online scenario short links from `🎵.example.com` would typically be
+redirected to `music.example.com`, such navigation redirection is not a good alternative with
+respect to offline scenarios. Therefore the app needs to register for both origins. {% endAside %}
+
+### The `web-app-origin-association` file
+
+Since the PWA lives on a different origin (`music.example.com`) than some of the URLs it needs to
+handle (e.g., `🎵.example.com`), the app needs to verify ownership of these other origins. This
+happens in a `web-app-origin-association` file hosted on said other origins.
+
+#### Structure of `web-app-origin-association` files
+
+This file must contain valid JSON. The top-level structure is an object, with a member named
+`"web_apps"`. This member is an array of objects and each object represents an entry for a unique
+web app. Each object contains:
+
+| Field        | Required | Description                                                          | Type     | Default |
+| ------------ | -------- | -------------------------------------------------------------------- | -------- | ------- |
+| `"manifest"` | yes      | URL string of the Web App Manifest of the associated PWA             | `string` | N/A     |
+| `"details"`  | no       | An object that contains arrays of included and excluded URL patterns | `object` | N/A     |
+
+Each `"details"` object contains:
+
+| Field             | Required | Description                      | Type       | Default |
+| ----------------- | -------- | -------------------------------- | ---------- | ------- |
+| `"paths"`         | no       | Array of allowed path strings    | `string[]` | `[]`    |
```

Comment:
Horizontal scrolling is not a best practice. Here's a way to compress this that's common in tech writing practice.

| Field    | Description |
| -------- | ------------ |
| `"Paths"` | (Required) An array of allowed path strings. The default is `[]`. |

Reply:
Using this trick now :-)